libs/agno/agno/agent/agent.py

```diff
@@ -740,6 +746,12 @@ def _run(
                 else:
                     self.run_response.tools.extend(model_response.tool_calls)
 
+                for tool_call in model_response.tool_calls:
+                    tool_name = tool_call.get("tool_name", "")
+                    if tool_name.lower() in ["think", "analyze"]:
```

Comment:
Does this cover ReasoningTools and ThinkingTools? And KnowledgeTools?

Reply:
not tested for `ThinkingTools? And KnowledgeTools?` yet